### ISSUE TITLE
Is this available for Kodi Nexus (v20)?

### ISSUE THREAD

**Issue by siliconhippy** [b-jesch/plugin.video.ipcams#11]
Hi there 😀

Great work...way lighter than Zone Minder when I used on Kodi Leia v18.

1. Is this also available for Kodi Nexus? There doesn't seem to be much change from Matrix....same Python 3?
2.  So this will only carry video stream, not 2 way audio ?

Cheers 😹

**Comment by b-jesch** [comment]
1. There are small changes to Python 3 (Matrix + Nexus) so this addon has to be adapted. Will make it soon.
2. The internal player of kodi takes the stream as it is, If there is audio included so the player will play it (if the stream is not as exotic).

**Comment by b-jesch** [comment]
Has just been updated for Kodi Matrix. Should run in Nexus too. Available at the Kodinerds repo: https://repo.kodinerds.net

**Comment by siliconhippy** [comment]
Thanks. That was quick !

Happy New Year, a bit late ! 😁

**Comment by siliconhippy** [comment]
The repo.kodinerds.net zip is still an old one that gives a 2.25 dependency error, like the master zip from this repo on my Kodi RC2 AE build 11.0.19 from AlexElec.tv

However the matrix zip from this repo installs correctly. But then it gives me an error upon Run.

See attached pic... 

I checked the URL of the Android IPWebcam app in Windows ONVIF Device Manager as: rtsp://192.168.0.26:8080/h264_ulaw.sdp
But feeding this URL in your Surveillance Cam addon still gives me same error below !

Line 42 in plugin.video.ipcam/default.py
No attribute 'translatePath'

![IMG_20230112_085311](https://user-images.githubusercontent.com/42253841/212086581-b7f37ab1-84eb-4542-892f-63338568189b.jpg)

**Comment by b-jesch** [comment]
Thats indeed a bug. Have overseen this while porting to Matrix. Will fix it immediately.

**Comment by b-jesch** [comment]
Fixed!

> However the matrix zip from this repo installs correctly.

However, for Matrix you have to choose the Matrix build. This is available at Kodinerds repo too. Maybe you have to update the repo.kodinerds.net (zip) to 7.0.1.2

**Comment by siliconhippy** [comment]
I downloaded the 7.0.12 Kodinerds repo zip ( the web page is outdated), and reinstalled your Surveillance Cam addon.

Now it says to define stream URL even though there is a URL definition available!
See pics...

( See edit below.)

![IMG_20230114_091513](https://user-images.githubusercontent.com/42253841/212476561-e824e705-29b7-456c-9bb0-575bc6c0edd9.jpg)

**Comment by siliconhippy** [comment]
I tried other several combinations suggested by Android IP Webcam ( that displays on my phone screen):
rtsp://192.168.0.34:8080/h264_ulaw.sdp
rtsp://192.168.0.34:8080/h264_pcm.sdp
http://192.168.0.34:8080/ONVIF/device_service
http://192.168.0.34:8080

![IMG_20230114_092736](https://user-images.githubusercontent.com/42253841/212477343-436e9922-2389-4e43-85aa-8fd5909a8517.jpg)

**Comment by b-jesch** [comment]
Fixed. Sorry

**Comment by siliconhippy** [comment]
Thanks for the quick fixes on the weekend!
German efficiency... 

Works with /h264_ulaw.sdp in 1280*1080.

But there is substantial (1+ sec) audio and video lag, and bottom half pixelation- is this the IP Webcam app or just Kodi streaming? 

Or maybe WiFi speed?  Mine is 30/15 Mbps down/upload but some occasional speed crashes... currently Speedtest.net in my phone browser is stuck at GO, which likely is a Chrome problem.

**Comment by b-jesch** [comment]
The app takes the stream as it is and playing it with the internal kodi player. You have to adjust the lag with the audio settings within the OSD settings menu. There's no other way (sorry for the german screenshot).

![Bildschirmfoto von 2023-01-15 12-37-46](https://user-images.githubusercontent.com/10152384/212538443-2ee392d7-3933-4ab7-9daf-7627fa6b090a.png)

**Comment by siliconhippy** [comment]
Ok so could be a Wifi speed slowdown issue too.
At least it works now 🐱

**Comment by siliconhippy** [comment]
Well a stream URL definition problem!

I see my INQMEGA FHD 1080P  WiFi camera being discovered as rtsp://198.168.0.27:554 with user/password as admin/ (none) in ONVIF Device Manager in Windows.

In VLC it shows as all formats:

A.  rtsp://admin:@198.168.0.27:554,
B. rtsp://admin:@198.168.0.27:554/h264_stream
C. rtsp://admin:@198.168.0.27:554/0/av0
D. rtsp://admin:@198.168.0.27
E. rtsp://admin:@198.168.0.27:8001, and :8001/h264_stream, and :8001/0/av0

These are all formats suggested online for INQMEGA model, while the INQMEGA ycc365plus.com "official" site suggests (A) only:

https://www.ycc365plus.com/en/ycc365-plus-camera-rtsp-url-address-and-port-number
https://camlytics.com/camera/inqmega

Unluckily I seem to have tried all these formats in this Nexus Surveillance Cam as above. 

I get a log error upon clicking Camera 1 menu in Run mode. But my kodi.log indicated nothing !

Any insights here?

**Comment by b-jesch** [comment]
Maybe create a .strm file and play it with the addon like this: https://www.arcdyn.com/articles/how-to-stream-a-security-camera-using-rtsp-kodi-on-raspberry-pi/

**Comment by siliconhippy** [comment]
Hi Birger !

Figured out the problem, but don't know if Kodi STRM will fix...

See Kodi log with debugging turned on:
https://paste.kodi.tv/azocekalom.kodi

I just used the .strm file from Video... Addons...Dir...File. I didn't use your addon: looks like it isn't your addon issue !

1. So I get errors (there was a previous session where I tried :554//0/av0, :8001/0/avo, :554 etc without luck), if I use the user/pwd formatting:
rtsp://admin:@IP:port/xyz

2. However I tried another way: Android IP WebCam app with no user/pwd, and it works nicely  as before, when you fixed the Nexus update of your addon:

rtsp://192.168.0.31:8080/h264_ulaw.sdp

3. I tried above URLs in .strm file for the INQMEGA cam without user/pwd, but there was no prompt for user/pwd input in Estuary Skin, and so it didn't work.

VLC, however prompts you for user/pwd if not already part of the URL.

4. So my conclusion:
Kodi .strm Player has an issue with blank passwords, or maybe user/passwords, in stream URL.

I cannot check because I don't know how to change password on my INQMEGA cam: poor doc, and no email reply yet from the Amazon Seller.)
